torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -2192,6 +2193,44 @@ int hashValue(Stack& stack) {
   return 0;
 }
 
+int64_t getIValueId(const IValue& val) {
+  // id in practice is only used to compare pointer types, and None.
+  // Python gives no guarantees otherwise.
+  if (val.isPtrType()) {
+    return reinterpret_cast<int64_t>(val.internalToPointer());
+  }
+  if (val.isNone()) {
+    return 0;
+  }
+  // For non-None / pointer types, make sure that they don't compute id(0)
+  if (val.isInt()) {
```

Comment:
Having this implemented for primitives is going to cause weird problems, as is won't `id(1) == id(True)`? I think we should only have `id()` implemented for pointer types (and this should happen at the schema level) since that's how people should be using it anyways.

Reply:
This was my original approach, updated